operatorPointwise_binary.go

```diff
@@ -422,26 +409,14 @@ func hadamardProdDiff(x, y, z *Node) (err error) {
 	mul := newEBOByType(mulOpType, ydv.Value.Type(), zdvdType)
 	err = mul.IncrDo(xdv.d, ydv.Value, zdv.d)
 	if err != nil {
-		var ver Valuer
-		var ok bool
-		if ver, ok = err.(Valuer); !ok {
-			return
-		}
-
-		xdv.SetDeriv(ver.Value()) // ignore errors on purpose
+		return errors.Wrap(err, "IncrDo call failed")
```

Comment:
This will cause A LOT of bugs. Don't do this. 

The main idea is this: there are two `Value` types: `Tensor` and `Scalar`.

`IncrDo` is the representation of ` x += (op)...`. It relies on the fact that Tensors are generally pointer types.

Now, you can't do the same on scalar types because they ain't pointers, so you have to wait for an error, see if it is a `Valuer`, and if it isn't then use `SetDeriv` to update the increased version